agent/runner/actions/mongodb_explain_action_test.go

```diff
@@ -33,6 +36,221 @@ import (
+			  "distinct": "inventory",
+			  "key": "dept",
+			  "query": {}
+			  },
+			  "$db": "testdb"
+			}
+		  }`
+		runExplain(ctx, t, prepareParams(t, query))
+	})
+
+	t.Run("Insert - not supported", func(t *testing.T) {
+		query := `{
+			"ns": "testdb.listingsAndReviews",
+			"op": "insert",
+			"command": {
+			  "insert": "listingsAndReviews",
+			  "ordered": true,
+			  "$db": "testdb"
+			}
+		  }`
+		runExplainExpectError(ctx, t, prepareParams(t, query))
+	})
+
+	t.Run("Drop - not supported", func(t *testing.T) {
+		query := `{
+			"ns": "testdb.listingsAndReviews",
+			"op": "command",
+			"command": {
+			  "drop": "listingsAndReviews",
+			  "$db": "testdb"
+			}
+		  }`
+		runExplainExpectError(ctx, t, prepareParams(t, query))
+	})
+
+	t.Run("PMM-12451", func(t *testing.T) {
+		// Query from customer to prevent wrong date/time, timestamp parsing in future and prevent regression.
+		query := `{"ns":"testdb.testDoc","op":"query","command":{"find":"testDoc","filter":{"$and":[{"c23":{"$ne":""}},{"c23":{"$ne":null},"delete":{"$ne":true}},{"$and":[{"c23":"985662747"},{"c15":{"$gte":{"$date":"2023-09-19T22:00:00.000Z"}}},{"c15":{"$lte":{"$date":"2023-10-20T21:59:59.000Z"}}},{"c8":{"$in":["X1118630710X","X1118630720X","X1118630730X","X1118630740X","X1118630750X","X1118630760X"]},"c22":{"$in":["X1118630710X","X1118630710XA","X1118630710XB","X1118630710XC","X1118630710XD","X1118630710XE"]},"c34":{"$in":["X1118630710X","X1118630710Y","X1118630710Z","X1118630710U","X1118630710V","X1118630710W"]}},{"c2":"xxxxxxx"},{"c29":{"$in":["X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X","X1118630710X"]}}]}]},"lsid":{"id":{"$binary":{"base64":"n/f5RI2jTTCoyt0y+8D9Cw==","subType":"04"}}},"$db":"testdb"}}`
+		runExplain(ctx, t, prepareParams(t, query))
+	})
+}
+
+func prepareParams(t *testing.T, query string) *agentpb.StartActionRequest_MongoDBExplainParams {
+	t.Helper()
+
+	return &agentpb.StartActionRequest_MongoDBExplainParams{
+		Dsn:   tests.GetTestMongoDBDSN(t),
+		Query: query,
+	}
+}
+
+func runExplain(ctx context.Context, t *testing.T, params *agentpb.StartActionRequest_MongoDBExplainParams) {
+	t.Helper()
+
+	big, err := rand.Int(rand.Reader, big.NewInt(27))
+	require.NoError(t, err)
+	id := strconv.FormatUint(big.Uint64(), 10)
+	ex, err := NewMongoDBExplainAction(id, 0, params, os.TempDir())
+	require.NoError(t, err)
+	res, err := ex.Run(ctx)
+	require.NoError(t, err)
+	require.NotEmpty(t, string(res))
```

Comment:
is there any standard thing that we can use to check that explain returns some logical response, not a random value.

Reply:
Main problem with this is warning in MongoDB doc:
```
WARNING
MongoDB does not guarantee any specific output format from the [explain] command, even when using the Stable API.
```

https://www.mongodb.com/docs/manual/reference/command/explain/

Right now we are testing the possibility to parse a query and run it without error. This should mean "correct" implementation (for specific query) on our side. Everything else comes from Mongo itself. If we are talking about checking something specific in response I think "winningPlan" should be ideal. But to have "winningPlan" not "EOF" we need to fulfil DB with data corresponding with every query in tests. I will verify this with Ivan, if it makes sense and what we should check. Once he will be back from vacation.